### ISSUE TITLE
Add an upgrade path from menu_block to core menu blocks. 

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#3681]
Add an upgrade path from menu_block to core menu blocks. 

Follow-up to https://github.com/backdrop-ops/backdropcms.org/issues/509 and https://github.com/backdrop/backdrop-issues/issues/3680

**Comment by jackaponte** [comment]
Hi Jen! Curious about whether you're planning work on this in the not-too-distant future?

**Comment by jackaponte** [comment]
Actually, just realized--this upgrade path would be for Drupal 7 sites using `menu_blocks` to upgrade to core menu blocks; it would _not_ cover migrating from `menu_blocks` to core in an existing Backdrop site, would it?

**Comment by jenlampton** [comment]
I don't know what you mean by "migrating" but if you mean using the migrate module, then no. I was specifically talking about what happens when you run `update.php` from Drupal 7, if the Drupal 7 site had `menu_block` module installed.

**Comment by jackaponte** [comment]
By "migrating" I meant an automated migration within an existing Backdrop installation from using the `menu_block` module to using Backdrop core menu blocks. I realize that the upgrade path in this ticket won't cover that! Thanks for the clarification. :)

**Comment by jenlampton** [comment]
I didn't know there was a Backdrop verison of menu_block...

**Comment by jackaponte** [comment]
> I didn't know there was a Backdrop verison of `menu_block` ... it's not in contrib. Are you sure that's the module you're using?

Yup, I've been using a _very_ unofficial port, https://github.com/docwilmot/menu_block, it's not even in https://github.com/backdrop-contrib! I used it on a site that we built more than two years ago, gotta finally move from that module to the core functionality.

**Comment by jenlampton** [comment]
well, if the settings are in the `variables` table the official D7 contrib -> B core update might still work. :) Want to send me a sanitized DB dump so I can give it a crack?

**Comment by jackaponte** [comment]
> Want to send me a sanitized DB dump so I can give it a crack?

Thank you for offering! I feel silly having to ask, but when you say "sanitized DB dump," do you just mean a DB dump from which I've removed all potentially sensitive info? If yes, how do you sanitize your DB dumps? Just manually removing the rows from the dump, or drop or truncate tables during the dump, or something more clever?

**Comment by jenlampton** [comment]
> how do you sanitize your DB dumps?

It depends on the site, as some have other places that contain sensitive information, but I'm always concerned about the users table.

I export a recent prod db and import it someplace temporary, either a separate db on the same server, or more often, a local db.

Then, my go-to is to use the `drush sql-san` command. If that doesn't work (usually when drush is not on whatever server I'm on), then I manually run the query `UPDATE users SET email = "a@b.c" password = MD5("password");` before I make a mysqldump :)
